src/helpers/transport/bytearrstream/mod.rs

```diff
@@ -0,0 +1,62 @@
+mod aligned;
+
+pub use aligned::ByteArrStream as AlignedByteArrStream;
+
+use crate::error::BoxError;
+use axum::extract::BodyStream;
+use futures::Stream;
+use futures_util::{stream, TryStreamExt};
+use hyper::body::Bytes;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// represents the item of an underlying stream
+type Item = Result<Bytes, BoxError>;
+/// represents the underlying stream
+type PinnedStream = Pin<Box<dyn Stream<Item = Item> + Send>>;
```

Comment:
apologies for bringing it late, but isn't it https://docs.rs/futures/latest/futures/stream/type.BoxStream.html/?